web/src/drafts.js

```diff
@@ -199,11 +199,9 @@ export function snapshot_message() {
         message.reply_to = recipient;
         message.private_message_recipient = recipient;
     } else {
-        message.stream = compose_state.stream_name();
-        const sub = stream_data.get_sub(message.stream);
-        if (sub) {
-            message.stream_id = sub.stream_id;
-        }
+        message.stream_id = compose_state.stream_id();
+        const sub = stream_data.get_sub_by_id(message.stream_id);
+        message.stream = sub ? sub.name : "";
```

Comment:
I agree that would be helpful! I'll note it as a followup investigation.

Reply:
Yeah the reason this was `stream_name` was just because it was a bare text input, so we had to be able to represent `core tea` if that's what was written in the compose box.

Needs some migration code to handle the old format (probably just dropping the value as unset if it's not a currently existing stream as a one-time migration thing) but this will be very nice to change.